### PR TITLE
torch: attention softmax as float32 for bf16

### PR DESCRIPTION
**Pull request by gyin94**

### CODE REVIEW

**Review by ruomingp — APPROVED** [review]
Is there a unittest that can be updated?

**Comment by gyin94** [comment]
@ruomingp good question. the unit test is running on CPU. May need to figure out how pytorch community is running bfloat16 for their implementation
